components/resourceDetails/resourceSharing/addAgentButton/addAgentRow/index.test.jsx

```diff
@@ -22,18 +22,22 @@
 import React from "react";
 import userEvent from "@testing-library/user-event";
 import { useSession, useThing } from "@inrupt/solid-ui-react";
-import { createThing } from "@inrupt/solid-client";
+import { addUrl, createThing } from "@inrupt/solid-client";
 import AddAgentRow from "./index";
 import { renderWithTheme } from "../../../../../__testUtils/withTheme";
+import useContactProfile from "../../../../../src/hooks/useContactProfile";
+import { vcardExtras } from "../../../../../src/addressBook";
 
 jest.mock("@inrupt/solid-ui-react");
+jest.mock("../../../../../src/hooks/useContactProfile");
```

Comment:
No, because we're mocking different resolved values in different describe blocks. In fact I had to put them in separate describe blocks (as opposed to nested describe blocks) because the mocks wouldn't be cleared in between tests.

Reply:
Yes, but we can still say `const mockedThingHook = useThing;` to group the mocked version of `useThing` together with `jest.mock("@inrupt/solid-ui-react");`. You would still use `mockReturnValue` in the various places, but doing `mockedThingHook.mockReturnValue` instead of `useThing.mockReturnValue`.